src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -28,6 +29,11 @@ namespace Microsoft.CodeAnalysis.CSharp
     internal abstract partial class Symbol : ISymbolInternal, IFormattable
     {
         private ISymbol _lazyISymbol;
+        // Cached kind
+        private SymbolKind _lazyKind = (SymbolKind)(-1);
+        private Symbol _lazyOriginalSymbolDefinition = null;
+        private ModuleSymbol _lazyContainingModule = null;
+        private Symbol _lazyContainingSymbol = null;
```

Comment:
similar concern here.  this seems to grow all symbols by 4 fields.  Do we know what impact that will have on things like IDE scenarios and whatnot?

Reply:
Agreed. I think we'll need to see benchmarks of memory pressure before and after making this change in order to consider such a change. We can have thousands to millions of symbols in memory at a time, meaning that this could have a _very_ outsized impact on memory pressure that will not be quickly cleaned up (as many symbols are shared from compilation to compilation).